src/main/java/com/google/googleidentity/servlet/RegisterCheckServlet.java

```diff
@@ -47,19 +48,32 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
       throws ServletException, IOException {
     String username = request.getParameter("username");
     String password = request.getParameter("password");
+    String email = request.getParameter("email");
+
+    if (Strings.isNullOrEmpty(username)
+        || Strings.isNullOrEmpty(password)
+        || Strings.isNullOrEmpty(email)) {
+      // This will not happen from the front end interface.
+      // Just avoid direct request error.
+      return;
+    }
 
     response.setContentType("text/html;charset=utf-8");
 
-    if (userDetailsService.getUserByName(username).isPresent()) {
+    if (userDetailsService.getUserByName(username).isPresent()
+        || userDetailsService.getUserByEmailOrGoogleAccountId(email, null).isPresent()) {
```

Comment:
I think we'd better split the check into two if statements, and return different error message to tell user which one  exists.